packages/Adapters/STKMesh/example/mesh_deformation.cpp

```diff
@@ -0,0 +1,414 @@
+
+        coords = stk::mesh::field_data( *coord_data, fixed_part_nodes[n] );
+        coord_field[f*3] = coords[0];
+        coord_field[f*3+1] = coords[1];
+    }
+
+    // Get the moving data.
+    for ( int n = 0; n < moving_num_part_nodes; ++n )
+    {
+        f = n + fixed_num_part_nodes;
+
+        displace_field[f*3] = ux;
+        displace_field[f*3+1] = uy;
+
+        coords = stk::mesh::field_data( *coord_data, moving_part_nodes[n] );
+        coord_field[f*3] = coords[0];
+        coord_field[f*3+1] = coords[1];
+    }
+    
+    // Get the interior data.
+    for ( int n = 0; n < num_interior; ++n )
+    {
+        f = n + num_boundary;
+
+        displace_field[f*3] = 0.0;
+        displace_field[f*3+1] = 0.0;
+
+        coords = stk::mesh::field_data( *coord_data, interior_part_nodes[n] );
+        coord_field[f*3] = coords[0];
+        coord_field[f*3+1] = coords[1];
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Deform the mesh.
+void deformMesh( 
+    stk::mesh::BulkData& bulk_data,
+    stk::mesh::Selector& fixed_selector,
+    stk::mesh::Selector& moving_selector,
+    stk::mesh::Selector& interior_selector,
+    Teuchos::Array<double>& displace_field )
+{
+    // Get the fixed boundary nodes.
+    stk::mesh::BucketVector fixed_part_buckets = 
+        fixed_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> fixed_part_nodes;
+    stk::mesh::get_selected_entities( 
+        fixed_selector, fixed_part_buckets, fixed_part_nodes );
+    int fixed_num_part_nodes = fixed_part_nodes.size();
+
+    // Get the moving boundary nodes.
+    stk::mesh::BucketVector moving_part_buckets = 
+        moving_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> moving_part_nodes;
+    stk::mesh::get_selected_entities( 
+        moving_selector, moving_part_buckets, moving_part_nodes );
+    int moving_num_part_nodes = moving_part_nodes.size();
+
+    // Get the number of boundary nodes.
+    int num_boundary_nodes = 
+        fixed_num_part_nodes + moving_num_part_nodes;
+
+    // Get the interior nodes.
+    stk::mesh::BucketVector interior_part_buckets = 
+        interior_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> interior_part_nodes;
+    stk::mesh::get_selected_entities( 
+        interior_selector, interior_part_buckets, interior_part_nodes );
+    int interior_num_part_nodes = interior_part_nodes.size();
+
+    // Get the coordinate field.
+    const stk::mesh::FieldBase* coord_field_base= 
+	bulk_data.mesh_meta_data().coordinate_field();
+    const stk::mesh::Field<double,stk::mesh::Cartesian3d>* coord_field =
+	dynamic_cast<const stk::mesh::Field<double,stk::mesh::Cartesian3d>* >(
+	    coord_field_base);
+    double* coords;
+
+    // Get the fixed boundary node coordinates.
+    int f = 0;
+    for ( int n = 0; n < fixed_num_part_nodes; ++n )
+    {
+        f = n;
```

Comment:
no but you need it in the rest of the loops over node blocks so i left it for consistency